File: Section-2/CodeSrc/Tree/BinaryTreeTilt.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.Design;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Utils.DS;

namespace CodeSrc.Tree
{
    class BinaryTreeTilt
    {
        public int FindTilt(TreeNode root)
        {
            int res = 0;
            if (root == null)
            {
                return 0;
            }

            res += CalculateTilt(root);
            res += FindTilt(root.left);
            res += FindTilt(root.right);

            return res;
        }

        private int CalculateTilt(TreeNode root)
        {
            return System.Math.Abs(CalculateSum(root.left) - CalculateSum(root.right));
        }

        private int CalculateSum(TreeNode root)
        {
            int sum = 0;
            if (root == null)
            {
                return 0;
            }

            sum += root.val;
            sum += CalculateSum(root.left);
            sum += CalculateSum(root.right);

            return sum;
        }
    }
}

File: src/main/java/lizbyu/binary/tree/BinaryTreeTilt.java
package lizbyu.binary.tree;

import lizbyu.utils.node.TreeNode;

public class BinaryTreeTilt {
    public int findTilt(TreeNode root) {
        if (root == null) {
            return 0;
        }

        return calculateTilt(root) + findTilt(root.left) + findTilt(root.right);
    }

    private int calculateTilt(TreeNode root) {
        return Math.abs(calculateSum(root.left) - calculateSum(root.right));
    }

    private int calculateSum(TreeNode root) {
        if (root == null) {
            return 0;
        }

        int sum = root.val;
        sum += calculateSum(root.left);
        sum += calculateSum(root.right);
        return sum;
    }
}
